### PR TITLE
feature/dashbord

### LINKED ISSUE #87: Dashboard for requester-user
as a requester/institution user i want to visit my dashboard and view and update my orders

layout:
https://drive.google.com/open?id=1UFBXLMwAhugQ09dC833elRGjsurMa_VE

### PR DESCRIPTION
**Pull request by emptycurse**
https://github.com/print4health/print4health/issues/87

### CODE REVIEW

File: assets/js/app.js [hunk 1]
```diff
@@ -6,13 +6,15 @@ import {
...
 } from 'react-router-dom';
+import PrivateRoute from './component/security/PricateRoute';
 import ReactGA from 'react-ga';
 import Index from './container/index/index';
 import UserNav from './component/user/user-nav';
 import ThingListContainer from './container/thing/list';
 import ThingDetailContainer from './container/thing/detail';
 import RequestPasswordResetModal from './component/modal/request-password-reset';
 import logo from '../logo-print4health-org.svg';
+import Dashbord from './container/dashboard/dashbord';
```
Comment (gherkins) [inline assets/js/app.js]: minor typo - should be "Dashboard"
```suggestion
import Dashboard from './container/dashboard/dashboard';
```

File: assets/js/app.js [hunk 2]
```diff
@@ -120,17 +131,16 @@ class App extends React.Component {
...
                   </Nav>
                 </Navbar.Collapse>
               </div>
             </Navbar>
           </header>
           <main className="container py-5">
-            <DismissableAlert message={this.state.alertMessage} variant={this.state.alertClass} />
+            <DismissableAlert message={alertMessage} variant={alertClass} />
             <Switch>
+              <PrivateRoute path="/dashbord" component={Dashbord} authed={user && Object.keys(user).length !== 0 && this.getCurrentUserRole() === "ROLE_REQUESTER"} user={user}/>
```
Comment (gherkins) [inline assets/js/app.js]: /dashboard :) 
```suggestion
              <PrivateRoute path="/dashboard" component={Dashboard} authed={user && Object.keys(user).length !== 0 && this.getCurrentUserRole() === "ROLE_REQUESTER"} user={user}/>
```

File: assets/js/component/security/PricateRoute.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Redirect, Route } from "react-router-dom";
+import React from "react";
+import PropTypes from "prop-types";
+
+PrivateRoute.propTypes = {
```
Comment (gherkins) [inline assets/js/component/security/PricateRoute.js]: Filename typo "PricateRoute

File: assets/js/component/user/user-nav.js [hunk 4]
```diff
@@ -39,11 +40,18 @@ class UserNav extends React.Component {
...
-      return <span>
-        <a href="#" className="nav-link" onClick={this.handleLogout}>
-          Abmelden
-        </a>
-      </span>;
+      return (
+        <React.Fragment>
+          {this.context.getCurrentUserRole() === 'ROLE_REQUESTER' &&
+          <li className="nav-item">
+            <NavLink className="nav-link" activeClassName="text-primary" to="/dashbord">Dashbord</NavLink>
```
Comment (gherkins) [inline assets/js/component/user/user-nav.js]:
```suggestion
            <NavLink className="nav-link" activeClassName="text-primary" to="/dashboard">Dashboard</NavLink>
```

File: assets/js/container/dashboard/dashbord.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import AppContext from "../../context/app-context";
+
+class Dashbord extends React.Component {
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
class Dashboard extends React.Component {
```
also in filename !

File: assets/js/container/dashboard/dashbord.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+  renderTale() {
+    const { data } = this.state;
+
+    if (!data || !data.orders) {
+      return;
+    }
+
+    return(
+      <div className="Dashbord__table">
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
      <div className="Dashboard__table">
```

File: assets/js/container/dashboard/dashbord.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        <div className="row Dashbord__headline-row">
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
        <div className="row Dashboard__headline-row">
```

File: assets/js/container/dashboard/dashbord.js [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+          <div className="col-md-2 Dashbord__headline">
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
          <div className="col-md-2 Dashboard__headline">
```

File: assets/js/container/dashboard/dashbord.js [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+            <h6>Produkt Bild</h6>
+          </div>
+          <div className="col-md-2 Dashbord__headline">
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
          <div className="col-md-2 Dashboard__headline">
```

File: assets/js/container/dashboard/dashbord.js [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+            <h6>Produkt Name</h6>
+          </div>
+          <div className="col-md-2 Dashbord__headline">
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
          <div className="col-md-2 Dashboard__headline">
```

File: assets/js/container/dashboard/dashbord.js [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+            <h6>Priorität Bestellung</h6>
+          </div>
+          <div className="col-md-2 Dashbord__headline">
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
          <div className="col-md-2 Dashboard__headline">
```

File: assets/js/container/dashboard/dashbord.js [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+            <h6>Bedarf</h6>
+          </div>
+          <div className="col-md-2 Dashbord__headline">
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
          <div className="col-md-2 Dashboard__headline">
```

File: assets/js/container/dashboard/dashbord.js [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+            <h6>Herstellerbestätigung</h6>
+          </div>
+          <div className="col-md-2 Dashbord__headline">
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
          <div className="col-md-2 Dashboard__headline">
```

File: assets/js/container/dashboard/dashbord.js [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+            <h6>Bereits gedruckt</h6>
+          </div>
+        </div>
+        {data.orders.map((order, i) => {
+          return (
+            <div key={`order_${i}`} className='row Dashbord__value-row'>
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
            <div key={`order_${i}`} className='row Dashboard__value-row'>
```

File: assets/js/container/dashboard/dashbord.js [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+              <div className="col-md-2">
+                <div className='Dashbord__value Dashbord__value-img'>
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
                <div className='Dashboard__value Dashboard__value-img'>
```

File: assets/js/container/dashboard/dashbord.js [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+                  {/* eslint-disable-next-line react/jsx-no-target-blank */}
+                  <a href={order.thing.url} target='_blank'>
+                    <img key={`things_image_${i}`} src={order.thing.imageUrl}/>
+                  </a>
+                </div>
+              </div>
+              <div className="col-md-2">
+                <div className='Dashbord__value'>
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
                <div className='Dashboard__value'>
```

File: assets/js/container/dashboard/dashbord.js [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+                  {/* eslint-disable-next-line react/jsx-no-target-blank */}
+                  <a href={order.thing.url} target='_blank'>
+                    <p>{order.thing.name}</p>
+                  </a>
+                </div>
+              </div>
+              <div className="col-md-2">
+                <div className='Dashbord__value'>
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
                <div className='Dashboard__value'>
```

File: assets/js/container/dashboard/dashbord.js [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+                  {
+                    // @TODO replace dummy value
+                  }
+                  <p>Normal</p>
+                </div>
+              </div>
+              <div className="col-md-2">
+                <div className='Dashbord__value'>
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
                <div className='Dashboard__value'>
```

File: assets/js/container/dashboard/dashbord.js [hunk 19]
```diff
@@ -0,0 +1,145 @@
...
+                  <p>{order.thing.needed}</p>
+                </div>
+              </div>
+              <div className="col-md-2">
+                <div className='Dashbord__value'>
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
                <div className='Dashboard__value'>
```

File: assets/js/container/dashboard/dashbord.js [hunk 20]
```diff
@@ -0,0 +1,145 @@
...
+                  {order.quantity - order.remaining}
+                </div>
+              </div>
+              <div className="col-md-2">
+                <div className='Dashbord__value'>
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
                <div className='Dashboard__value'>
```

File: assets/js/container/dashboard/dashbord.js [hunk 21]
```diff
@@ -0,0 +1,145 @@
...
+        })}
+      </div>
+    );
+  }
+
+  render() {
+    const { user } = this.context;
+
+    return (
+      <div className="container Dashbord">
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
      <div className="container Dashboard">
```

File: assets/js/container/dashboard/dashbord.js [hunk 22]
```diff
@@ -0,0 +1,145 @@
...
+        <div className="row">
+          <div className="col">
+            <h1>Dashbord</h1>
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
            <h1>Dashboard</h1>
```

File: assets/js/container/dashboard/dashbord.js [hunk 23]
```diff
@@ -0,0 +1,145 @@
...
+          </div>
+        </div>
+        <div className="row">
+          <div className="col Dashbord__user">
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
          <div className="col Dashboard__user">
```

File: assets/js/container/dashboard/dashbord.js [hunk 24]
```diff
@@ -0,0 +1,145 @@
...
+            <h6>{user.email}</h6>
+          </div>
+        </div>
+        {this.renderTale()}
+      </div>
+    )
+  }
+}
+
+Dashbord.contextType = AppContext;
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
Dashboard.contextType = AppContext;
```

File: assets/js/container/dashboard/dashbord.js [hunk 25]
```diff
@@ -0,0 +1,145 @@
...
+
+export default Dashbord;
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]:
```suggestion
export default Dashboard;
```

File: assets/scss/partials/_components.scss [hunk 26]
```diff
@@ -64,3 +64,81 @@
         margin: 0 auto;
     }
 }
+
+.Dashbord {
```
Comment (gherkins) [inline assets/scss/partials/_components.scss]:
```suggestion
.Dashboard {
```

File: assets/js/component/security/PricateRoute.js [hunk 27]
```diff
@@ -0,0 +1,22 @@
...
+  location: PropTypes.any,
+  authed: PropTypes.bool
+};
+
+function PrivateRoute({component: Component, authed}) {
+  return (
+    <Route
+      render={(props) => authed === true
+        ? <Component {...props} />
+        //@TODO display a flashmessage and change redirect to Login when we have a Login Page
```
Comment (gherkins) [inline assets/js/component/security/PricateRoute.js]: flashmessage could be done with `appContext.setAlert()`

File: assets/js/container/dashboard/dashbord.js [hunk 28]
```diff
@@ -0,0 +1,145 @@
...
+  async fetchData() {
+    const { user } = this.context;
+    const { data } = this.state;
+
+    if (data !== null) {
+      return;
+    }
+
+    try {
+      const response = await fetch(`/orders/requester/${user.id}`);
```
Comment (gherkins) [inline assets/js/container/dashboard/dashbord.js]: in other components we used the axios client 🤔 
maybe that's overhead then?

**Review by gherkins — CHANGES_REQUESTED** [review]
nice 💯 👍 🥇 

just some typo fixes and questions...

**Comment by bartrail** [comment]
typos due to maximum typing speed :D

**Review by gherkins — APPROVED** [review]

**Pull request merged**